wicket-core/src/main/java/org/apache/wicket/page/PageAccessSynchronizer.java

```diff
@@ -93,8 +94,8 @@ public Duration getTimeout(int pageId)
 	 */
 	public void lockPage(int pageId) throws CouldNotLockPageException
 	{
-		final Thread thread = Thread.currentThread();
-		final PageLock lock = new PageLock(pageId, thread);
+		final RequestCycle cycle = RequestCycle.get();

```

Comment:
Not true!

As long as one RequestCycle holds the lock on a page, no other RequestCycle can get a lock. It doesn't make any difference whether we lock on the thread or the RequestCycle.